File: clients/dart/tanks/my_tank.dart
import 'package:code_tanks/code_tanks.dart';

class MyTank extends BaseTank {
  MyTank() {
    print("Created my tank!");
  }

  @override
  void run() {
    // TODO: implement run
  }
  
  @override
  void onEvent(EventType e, Map info) {
    // TODO: implement onEvent
  }
}

BaseTank createTank() => MyTank();

File: runner/src/my_tank.rs
use ct_api::*;
use serde_json::Value;

pub struct MyTank {}

impl Tank for MyTank {
    fn run(&mut self, _commands: &mut Vec<Command>) {
        todo!()
    }

    fn on_event(&mut self, _commands: &mut Vec<Command>, _event: &Value) {
        todo!()
    }
}

pub fn create_tank() -> MyTank {
    MyTank {}
}
